src/EcoSurveyComponent.java

```diff
@@ -0,0 +1,147 @@
+import components.map.Map;
+import components.map.Map1L;
+import components.simplereader.SimpleReader;
+import components.simplereader.SimpleReader1L;
+import components.simplewriter.SimpleWriter;
+import components.simplewriter.SimpleWriter1L;
+
+public class EcoSurveyComponent {
+    private final class Species {
+
+        //Should these fields be public?
+        /**
+         * colloquial name
+         */
+        private String name;
+        /**
+         * scientific name
+         */
+        private String trueName;
+        /**
+         * ideal population density
+         */
+        private double density;
+
+    }
+
+    //is this redundant. Should I just create another private field for count?
+    private Map<Species, Integer> speciesCount;
+
+    private double area;
+
+    private int totalSpecies;
+
+    private void createNewRep() {
+        this.totalSpecies = 0;
+        this.area = 0;
+        this.speciesCount = new Map1L<>();
+    }
+
+    public EcoSurveyComponent() {
+        this.createNewRep();
+    }
+
+    //private method
+    private boolean hasSpecies(String name) {
+        //this function seems all wrong
+        boolean speciesFound = false;
+        for (Map.Pair<Species, Integer> s : this.speciesCount) {
+            if (s.key().name == name) {
+                speciesFound = true;
+            }
+        }
+        return speciesFound;
+    }
+
+    //I think this is a kernel function. I do want client to use it.
+    //Is it ok if it accesses fields/representation.
+    private void addSpecies(String name, String trueName, double density) {
+        assert !this.hasSpecies(name) : "Violation of species does not exist";
+
+        Species s1 = new Species();
+        s1.name = name;
+        s1.trueName = trueName;
+        s1.density = density;
+        this.speciesCount.add(s1, 0);
+        this.totalSpecies++;
+    }
+
+    //kernel. Should these methods be public?
+    private void setCount(String name, int count) {
```

Comment:
Why not? They provide convenience to the user. I know I generally would keep as many methods hidden as possible, but if this is useful to the client and would not cause them to hurt themselves, feel free to expose it. If these are kernel methods, keeping them private make sense as a way of showing the layering.